rust/abacus-core/src/types/chain_data.rs

```diff
@@ -0,0 +1,57 @@
+use ethers::prelude::{H256, U256};
+
+/// Info about a given block in the chain.
+#[derive(Debug)]
+pub struct BlockInfo {
+    /// Hash of this block
+    pub hash: H256,
+    /// Unix timestamp of this block's creation in seconds
+    pub timestamp: u64,
+    /// Block height or the nth block in the chain
+    pub number: u64,
+    /// Total used gas by transactions in this block
+    pub gas_used: U256,
+    /// Maximum amount of gas allowed in this block
+    pub gas_limit: U256,
+}
+
+/// Information about a given transaction in the chain.
+#[derive(Debug)]
+pub struct TxnInfo {
+    /// Hash of this transaction
+    pub hash: H256,
+    /// Amount of gas which was allocated for running the transaction
+    pub gas_limit: U256,
+    /// Represents the maximum tx fee that will go to the miner as part of the
+    /// user's fee payment.
+    pub max_priority_fee_per_gas: Option<U256>,
```

Comment:
How are you thinking about these structs in a heterogeneous execution environment world? Some of these fields are very specific to ethereum or ethereum-like chains.